packages/components/src/utils/hooks/test/use-controlled-value.js

```diff
@@ -0,0 +1,101 @@
+/**
+ * External dependencies
+ */
+import { fireEvent, render, screen } from '@testing-library/react';
+
+/**
+ * Internal dependencies
+ */
+import { useControlledValue } from '../use-controlled-value';
+
+function Input( props ) {
+	const [ value, setValue ] = useControlledValue( props );
+	return (
+		<input
+			value={ value }
+			onChange={ ( event ) => setValue( event.target.value ) }
+		/>
+	);
+}
+
+function getInput() {
+	return screen.getByRole( 'textbox' );
+}
+
+describe( 'useControlledValue', () => {
+	it( 'should use the default value', () => {
+		render( <Input defaultValue="WordPress.org" /> );
+		expect( getInput() ).toHaveValue( 'WordPress.org' );
+	} );
+
+	it( 'should use the default value then switch to the controlled value', () => {
+		const { rerender } = render( <Input defaultValue="WordPress.org" /> );
+		expect( getInput() ).toHaveValue( 'WordPress.org' );
+
+		rerender(
+			<Input defaultValue="WordPress.org" value="Code is Poetry" />
+		);
+		expect( getInput() ).toHaveValue( 'Code is Poetry' );
+	} );
+
+	it( 'should not call onChange only when there is no value being passed in', () => {
+		const onChange = jest.fn();
+		render( <Input defaultValue="WordPress.org" onChange={ onChange } /> );
+
+		expect( getInput() ).toHaveValue( 'WordPress.org' );
+
+		fireEvent.change( getInput(), { target: { value: 'Code is Poetry' } } );
+
+		expect( getInput() ).toHaveValue( 'Code is Poetry' );
+		expect( onChange ).not.toHaveBeenCalled();
+	} );
+
+	it( 'should call onChange when there is a value passed in', () => {
+		const onChange = jest.fn();
+		const { rerender } = render(
+			<Input
+				defaultValue="WordPress.org"
+				value="Code is Poetry"
+				onChange={ onChange }
+			/>
+		);
+
+		expect( getInput() ).toHaveValue( 'Code is Poetry' );
+
+		fireEvent.change( getInput(), {
+			target: { value: 'WordPress rocks!' },
+		} );
+
+		rerender(
```

Comment:
Do we need to serenader in order for the onChange function to get called?

Reply:
I believe so.

Reply:
We don't need to rerender for the `onChange` function to get called. It's triggered by the `fireEvent.change` call above that. But our `onChange` mock function is just `jest.fn()`. It doesn't really update any state as we would usually do in a component. So the rerender is updating the `value` to reflect the change. We could write that in a better way though.